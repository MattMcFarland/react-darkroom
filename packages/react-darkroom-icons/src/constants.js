export const iconKinds = {
  FLOPPYDISK: 'floppyDisk',
  UNDO: 'undo',
  UNDO2: 'undo2',
  REDO: 'redo',
  REDO2: 'redo2',
  CROP: 'crop',
  IMAGE: 'image',
  CROSS: 'cross',
  CHECKMARK: 'checkmark'
}
