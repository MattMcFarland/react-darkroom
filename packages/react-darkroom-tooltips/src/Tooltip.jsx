import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';
import {
  compose,
  pure,
  setDisplayName,
} from 'recompose';

import CARDINAL_DIRECTIONS from './constants';
import styles from './styles';

export const Tooltip = ({
  position,
  classes,
  children,
  label,
}) =>
  (<span className={`${classes[`tooltip-${position}`]} ${classes.base} `} data-label={label}>{children}</span>);

Tooltip.propTypes = {
  /**
   * The message displayed in the tooltip
   */
  label: PropTypes.string.isRequired,
  /** Determines the position the tooltip appears from, 
   * 
   * One of the following: `nw, n, ne, e, se, s, sw, w` */
  position: PropTypes.oneOf(Object.values(CARDINAL_DIRECTIONS)),
  /** @ignore JSS */
  classes: PropTypes.shape({
    'tooltip-nw': PropTypes.string,
    'tooltip-n': PropTypes.string,
    'tooltip-ne': PropTypes.string,
    'tooltip-e': PropTypes.string,
    'tooltip-se': PropTypes.string,
    'tooltip-s': PropTypes.string,
    'tooltip-sw': PropTypes.string,
    'tooltip-w': PropTypes.string,
  }).isRequired,
  /** Element to which the tooltip belongs, like a `<Button>` */
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
  /* eslint react/no-unused-prop-types: 0 */
  radius: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  backgroundColor: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  arrowSize: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

Tooltip.defaultProps = {
  position: 's',
  radius: '0.3em',
  backgroundColor: '#333',
  arrowSize: '0.5em',
};

export default compose(
  pure,
  injectSheet(styles),
  setDisplayName('Tooltip'),
)(Tooltip);
