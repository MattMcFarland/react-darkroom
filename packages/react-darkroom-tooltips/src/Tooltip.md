Tooltips wrap a single node, commonly a `button`.

### Default tooltip:
    <section>
      <Tooltip label="hello thar"><button>hover me...</button></Tooltip>
    </section>

### Positional tooltips:
You can set the position of any tooltip by using the `position` prop
Hover the text in **bold** below:

    <p>
      Lorem ipsum dolor sit 
      <Tooltip position="nw" label="foo bar"><b>amet</b></Tooltip>, consectetur 
      <Tooltip position="n" label="foo bar"><b>adipiscing</b></Tooltip> elit. Phasellus eget enim sit amet ipsum lobortis 
      <Tooltip position="ne" label="foo bar"><b>ultricies</b></Tooltip>. Etiam dictum arcu eget porta sagittis. Mauris tempus  
      <Tooltip position="e" label="foo bar"><b>volutpat</b></Tooltip> tellus vel pulvinar. 
      <Tooltip position="se" label="foo bar"><b>In</b></Tooltip> laoreet non libero vel accumsan. 
      <Tooltip position="s" label="foo bar"><b>Maecenas</b></Tooltip> tristique tortor id lacus tempor, 
      <Tooltip position="sw" label="foo bar"><b>eget</b></Tooltip> tincidunt risus bibendum. Ut vel rutrum lorem, sit amet vestibulum turpis. 
      <Tooltip position="w" label="foo bar"><b>Donec</b></Tooltip> feugiat auctor nunc, id suscipit quam luctus at. 
    </p>

### Customizing tooltips:
*advanced feature*

You can customize the appearance of tooltips by using the `TooltipFactory`

    // import { TooltipFactory } from 'react-darkroom-tooltips' // es6
    const TooltipFactory = require('./Tooltip').TooltipFactory; // es5

    // configure your tooltip with ToolTipFactory
    const MyTooltip = TooltipFactory({
      radius: '0.9em',
      color: '#ba4333',
      arrowSize: '0.5em',
    })
    
    // In your components, you can now use your tooltip the same way, but it
    // now has your new style settings!
    class Example extends React.Component {
      render() {
        return (
          <div>
            <MyTooltip position="e" label="niiice">Hover me...</MyTooltip>
          </div>
        );
      }
    }

    <Example/>
    
