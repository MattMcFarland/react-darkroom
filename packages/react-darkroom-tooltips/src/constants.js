export default ({
  NW: 'nw',
  N: 'n',
  NE: 'ne',
  E: 'e',
  SE: 'se',
  S: 's',
  SW: 'sw',
  W: 'w',
});
