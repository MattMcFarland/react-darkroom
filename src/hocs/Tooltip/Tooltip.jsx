import React from 'react'
import { compose } from 'recompose'
import injectSheet from 'react-jss'

