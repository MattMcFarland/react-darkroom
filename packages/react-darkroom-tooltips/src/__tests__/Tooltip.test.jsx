import React from 'react';
import renderer from 'react-test-renderer';

import Tooltip from '../Tooltip';
import constants from '../constants';

const { describe, test, expect } = global;

const createTests = positions =>
  Object.entries(positions).map(([key, value]) =>
    ({
      constantKey: key,
      mapsTo: value,
      node: (<Tooltip key={key} label="test" position={value} >test</Tooltip>),
    }));

describe('Tooltip', () => {
  const toolTipTests = createTests(constants);
  describe('constants are used to map to position props', () => {
    toolTipTests.forEach(({ constantKey, mapsTo, node }) => {
      test(`${constantKey} maps to position: ${mapsTo}`, () => {
        expect(renderer.create(node)).toMatchSnapshot();
      });
    });
  });
});

