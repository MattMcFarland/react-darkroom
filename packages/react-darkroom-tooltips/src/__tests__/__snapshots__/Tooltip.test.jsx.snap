// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip TooltipFactory Creates a tooltip  1`] = `
<span
  className="tooltip-s-0-13"
  data-label="test"
>
  hai
</span>
`;

exports[`Tooltip constants are used to map to position props E maps to position: e 1`] = `
<span
  className="tooltip-e-0-3"
  data-label="test"
>
  test
</span>
`;

exports[`Tooltip constants are used to map to position props N maps to position: n 1`] = `
<span
  className="tooltip-n-0-1"
  data-label="test"
>
  test
</span>
`;

exports[`Tooltip constants are used to map to position props NE maps to position: ne 1`] = `
<span
  className="tooltip-ne-0-2"
  data-label="test"
>
  test
</span>
`;

exports[`Tooltip constants are used to map to position props NW maps to position: nw 1`] = `
<span
  className="tooltip-nw-0-0"
  data-label="test"
>
  test
</span>
`;

exports[`Tooltip constants are used to map to position props S maps to position: s 1`] = `
<span
  className="tooltip-s-0-5"
  data-label="test"
>
  test
</span>
`;

exports[`Tooltip constants are used to map to position props SE maps to position: se 1`] = `
<span
  className="tooltip-se-0-4"
  data-label="test"
>
  test
</span>
`;

exports[`Tooltip constants are used to map to position props SW maps to position: sw 1`] = `
<span
  className="tooltip-sw-0-6"
  data-label="test"
>
  test
</span>
`;

exports[`Tooltip constants are used to map to position props W maps to position: w 1`] = `
<span
  className="tooltip-w-0-7"
  data-label="test"
>
  test
</span>
`;
